server/status.go

```diff
@@ -240,9 +357,31 @@ func (s *statusServer) handleLocalLogFile(w http.ResponseWriter, r *http.Request
 		return
 	}
 	w.Header().Set(util.ContentTypeHeader, contentType)
+	w.WriteHeader(http.StatusOK)
 	w.Write(b)
 }
 
+// handleLogFile handles GET requests for a single log file.
+func (s *statusServer) handleLogFile(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	nodeID, local, err := s.extractNodeID(ps)
+	if err != nil {
+		http.Error(w,
+			fmt.Sprintf("node id could not be parsed: %s", err),
+			http.StatusBadRequest)
+		return
+	}
+
+	if local {
+		s.handleLogFileLocal(w, r, ps)
+	} else {
+		file := ps.ByName("file")
+		// File cannot be empty here or else the request would have been routed
+		// to handleLogFilesList instead.
+		url := fmt.Sprintf("%s/%s", statusLogFilesListLocal, file)
```

Comment:
This is kind of weird - this is using 'statusLogFilesListLocal', but is giving a specific pattern.